src/runtime_src/core/pcie/emulation/common_em/config.h

```diff
@@ -89,6 +90,89 @@ namespace xclemulation{
   const int VIVADO_MIN_VERSION = 2000;
   const int VIVADO_MAX_VERSION = 2100;
 
+//sparse log utility
+enum class eEmulationType{
+  eSw_Emu,
+  eHw_Emu
+};
+
+struct sParseLog
+{
+  std::ifstream mFileStream;
+  std::string mFileName;
+  std::atomic_bool mFileExists;
+  std::vector<std::string> mMatchedStrings;
+  eEmulationType mEmuType;
+
+  sParseLog(const std::string& iDeviceLog, eEmulationType iType, const std::vector<std::string>& iMatchedStrings)
+      : mFileName(iDeviceLog)
+      , mFileExists{false}
+      , mMatchedStrings(iMatchedStrings)
+      , mEmuType(iType)
+  {
+
+  }
+
+  void check_simulator_status()
+  {
+    std::string line;
+    while (std::getline(mFileStream, line))
+    {
+      for (const auto &StringData : mMatchedStrings)
+      {
+        if (line.find(StringData) != std::string::npos)
+        {
+          if (eEmulationType::eSw_Emu == mEmuType)
+            print_user_msg();
+          else if (eEmulationType::eHw_Emu == mEmuType)
+          {
+            if (!StringData.compare("Exiting xsim") || !StringData.compare("FATAL_ERROR"))
```

Comment:
Is this an obfuscation contest? Just try a random understanding test with your colleagues :-)
```suggestion
            if (StringData == "Exiting xsim" || StringData == "FATAL_ERROR")
```

Reply:
Agree. This looks more cleaner.